### PR TITLE
Meta Audience Network Mediation Adapter 

### PR DESCRIPTION
**Pull request by LTPhantom**
## Description

Adds a new Flutter plugin that enables the Meta Audience Network Mediation Adapter to use with the google_mobile_ads flutter plugin.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_mobile_ads/mediation/gma_mediation_meta/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_meta/GmaMediationMetaPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.flutter.plugins.googlemobileads.mediation.gma_mediation_meta
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+
+/** GmaMediationMetaPlugin */
```
Comment (jjliu15) [inline packages/google_mobile_ads/mediation/gma_mediation_meta/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_meta/GmaMediationMetaPlugin.kt:19]: Nit: ending period. This comment also doesn't add any additional context/information.

**Review by jjliu15 — COMMENTED** [review]
Is this PR intended just to be the skeleton without any implementation? If so could you add that to the PR description?

**Comment by LTPhantom** [comment]
> Is this PR intended just to be the skeleton without any implementation? If so could you add that to the PR description?

The plugin is able to load ads from Meta by only linking the adapter dependency from the platform layer. Since no third party sdk or Mediation Extra is available, for now it will remain mostly empty.

**Review by luo86 — APPROVED** [review]

File: .github/workflows/gma_mediation_meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+on:
+  pull_request:
+    paths:
+    - "packages/mediation/gma_mediation_meta/**"
+    - ".github/workflows/gma_mediation_meta.yaml"
+    - "packages/mediation/gma_mediation_meta/ios/**"
+    - "packages/mediation/gma_mediation_meta/android/**"
+  push:
+    branches:
+    - master
```
Comment (jjliu15) [inline .github/workflows/gma_mediation_meta.yaml:26]: It looks like we never updated this to main. Also applies to other yaml files

File: analysis_options.yaml [hunk 3]
```diff
@@ -15,7 +15,10 @@
 include: package:pedantic/analysis_options.1.11.0.yaml
 
 analyzer:
-  exclude: [packages/gma_mediation_applovin/**]
+  exclude: [
+    packages/mediation/gma_mediation_applovin/**,
```
Comment (jjliu15) [inline analysis_options.yaml:19]: Why do we need this?

File: packages/mediation/gma_mediation_meta/example/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# gma_mediation_meta_example
+
+Demonstrates how to use the gma_mediation_meta plugin.
```
Comment (jjliu15) [inline packages/mediation/gma_mediation_meta/example/README.md:3]: Could you update the comment to explain that all that's needed is linking `gma_mediation_meta` and `google_mobile_ads` in your pubspec.yaml?

**Review by jjliu15 — APPROVED** [review]

File: .github/workflows/gma_mediation_meta.yaml (hunk 2 above)
Comment (LTPhantom) [reply .github/workflows/gma_mediation_meta.yaml]: Good catch, will fix for other yamls in a future PR

File: packages/mediation/gma_mediation_meta/example/README.md (hunk 4 above)
Comment (LTPhantom) [reply packages/mediation/gma_mediation_meta/example/README.md]: Added more info and link to developer's guide.

File: analysis_options.yaml (hunk 3 above)
Comment (LTPhantom) [reply analysis_options.yaml:19]: Each plugin has its own analysis_options file, that it is used by each workflow yaml to analyze individually when there is a change in the files. If this analysis file is left to check every folder, there are some unexpected problems with the dependencies and other analysis files.

**Pull request merged**